src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageAzureUtilityImpl.java

```diff
@@ -0,0 +1,283 @@
+package ca.corefacility.bioinformatics.irida.repositories.filesystem;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Optional;
+import java.util.zip.GZIPInputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import ca.corefacility.bioinformatics.irida.exceptions.StorageException;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
+import ca.corefacility.bioinformatics.irida.ria.web.dto.IridaTemporaryFile;
+import ca.corefacility.bioinformatics.irida.util.FileUtils;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.blob.models.BlobStorageException;
+
+/**
+ * Component implementation of file utitlities for azure storage
+ */
+@Component
+public class IridaFileStorageAzureUtilityImpl implements IridaFileStorageUtility {
+	private static final Logger logger = LoggerFactory.getLogger(IridaFileStorageAzureUtilityImpl.class);
+
+	private BlobServiceClient blobServiceClient;
+	private BlobContainerClient containerClient ;
+
+	@Autowired
+	public IridaFileStorageAzureUtilityImpl(String connectionStr, String containerName){
+		this.blobServiceClient = new BlobServiceClientBuilder().connectionString(connectionStr)
+				.buildClient();
+		this.containerClient = blobServiceClient.getBlobContainerClient(containerName);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public IridaTemporaryFile getTemporaryFile(Path file) {
+		// We set the blobClient "path" to which file we want to get
+		BlobClient blobClient = containerClient.getBlobClient(getAzureFileAbsolutePath(file));
+
+		try {
+			logger.trace("Getting file from azure [" + file.toString() + "]");
+			Path tempDirectory = Files.createTempDirectory(null);
```

Comment:
Can we add a prefix here instead of `null`?  That will make things easier to debug if something is going wrong leaving temp directories around.  Maybe something like `"azure-tmp-"`.  We could add similar for the S3 stuff in its PR.

Reply:
Sure good idea! Added in 117eedc7f06e586f08eed0fca9d520de62ca3c6f